### PR TITLE
Convert migration and cache tests to JUNIT4

### PR DESCRIPTION
**Pull request by beeender**

### CODE REVIEW

File: realm/realm-library/src/androidTest/java/io/realm/RealmMigrationTests.java [hunk 1]
```diff
@@ -30,24 +38,38 @@
...
-    public Realm realm;
+    private Realm realm;
+    private Context context;
 
-    @Override
-    protected void tearDown() throws Exception {
-        super.tearDown();
+    @Before
+    public void setup() {
+        context = InstrumentationRegistry.getInstrumentation().getContext();
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/RealmMigrationTests.java]: This context doesn't look like it is getting used?

File: realm/realm-library/src/androidTest/java/io/realm/rule/TestRealmConfigurationFactory.java [hunk 2]
```diff
@@ -109,4 +116,20 @@ public RealmConfiguration createConfiguration(String name, byte[] key) {
     public RealmConfiguration.Builder createConfigurationBuilder() {
         return new RealmConfiguration.Builder(getRoot());
     }
+
+    // Copies a Realm file from assets to temp dir
+    public void copyRealmFromAssets(Context context, String realmPath, String newName)
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/rule/TestRealmConfigurationFactory.java]: This is also in `TestHelper`, but I agree it makes more sense here. Can we delete it outright from `TestHelper` or do we need to deprecate it first until all classes have moved to use `TestRealmConfigurationFactory`?

Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/rule/TestRealmConfigurationFactory.java]: This is also in `TestHelper`, but I agree it makes more sense here. Can we delete it outright from `TestHelper` or do we need to deprecate it first until all classes have moved to use `TestRealmConfigurationFactory`?

**Comment by cmelchior** [comment]
Apart from a small comment :+1:

File: realm/realm-library/src/androidTest/java/io/realm/rule/TestRealmConfigurationFactory.java (hunk 2 above)
Comment (beeender) [inline realm/realm-library/src/androidTest/java/io/realm/rule/TestRealmConfigurationFactory.java]: The one in TestHelper is being used IOSRealmTest. No need to bother mark it as deprecated. I will just convert that on later today, just a few lines of code.

Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/rule/TestRealmConfigurationFactory.java]: :+1:

Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/rule/TestRealmConfigurationFactory.java]: :+1:

**Comment by stk1m1** [comment]
:+1: I've found few tricks I could use from the conversion. Thanks!

**Comment by zaki50** [comment]
:+1:

**Pull request merged**